### PR TITLE
windows CUDA support

### LINKED ISSUE #403: Ollama Windows version
I saw it is coming but didn't mention when? Would be great if you pinned this issue as more people use Windows & ollama has such a great dx.

The project looks absolutely brilliant. Would love to use text (gpt-4) & code (copilot) locally.

### PR DESCRIPTION
**Pull request by vinjn**
Fix #403 
- Support cuda build in Windows
- Import "containerd/console" lib to support colorful output in Windows terminal

image

### CODE REVIEW

**Review by skydiver1980 — APPROVED** [review]

File: cmd/cmd.go [hunk 1]
```diff
@@ -896,6 +898,11 @@ func checkServerHeartbeat(_ *cobra.Command, _ []string) error {
 func NewCLI() *cobra.Command {
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 
+	if runtime.GOOS == "windows" {
```
Comment (BruceMacD) [inline cmd/cmd.go:901]: This looks like an aesthetic change unrelated to the CUDA support, correct? There's gonna be a few things we need to fix in the CLI for an official windows release, so I think this would make more sense as a separate PR.

File: llm/llama.go [hunk 2]
```diff
@@ -84,6 +84,7 @@ func chooseRunners(workDir, runnerType string) []ModelRunner {
 	case "windows":
 		// TODO: select windows GPU runner here when available
 		runners = []ModelRunner{
```
Comment (BruceMacD) [inline llm/llama.go:86]: In this file you will also want to enable dynamic NumGPU allocation in the NumGPU function by doing the VRAM check in the case of windows also. Something like this:
```
index 7172f91..85540e8 100644
--- a/llm/llama.go
+++ b/llm/llama.go
@@ -269,7 +269,7 @@ func NumGPU(numLayer, fileSizeBytes int64, opts api.Options) int {
        if opts.NumGPU != -1 {
                return opts.NumGPU
        }
-       if runtime.GOOS == "linux" {
+       if runtime.GOOS == "linux" || runtime.GOOS == "windows" {
                freeBytes, err := CheckVRAM()
                if err != nil {
                        if !errors.Is(err, errNvidiaSMI) {
```

**Review by BruceMacD — COMMENTED** [review]
Thanks for your contribution and expertise, this is helpful. 

Couple of suggestions before we move forward, I'm also going to make the title of the PR more specific because there are some experience things we need to tweak in Ollama before an official Windows release.

File: llm/llama.go (hunk 2 above)
Comment (vinjn) [reply llm/llama.go:86]: Fixed, thx

**Review by BruceMacD — APPROVED** [review]
thanks again, merging this now
